build/dsc/templates/dsc_type.rb.erb

```diff
@@ -4,6 +4,7 @@ Puppet::Type.newtype(:dsc_<%= resource.friendlyname.downcase %>) do
   require Pathname.new(__FILE__).dirname + '../../' + 'puppet/type/base_dsc'
 
   provide :powershell, :parent => Puppet::Type.type(:base_dsc).provider(:powershell) do
+    confine :operatingsystem => :windows, :powershell_version => '5.0.10514.6'
```

Comment:
Also, I should have elaborated on the confine comparison @ferventcoder.  Thanks for pointing that out.  We do need to rewrite the confine a bit since the current fact comparison is a match.  Assuming the fact can return a `Gem::Version`, then this:

```ruby
confine :true => (Facter[:powershell_version] >= Gem::Version.new('5.0.10240.16384'))
```

If its a string, then:

```ruby
confine :true => (Gem::Version.new(Facter[:powershell_version]) >= Gem::Version.new('5.0.10240.16384'))
```

Hope thats clearer.